services/postgres/src/db_export.ts

```diff
@@ -0,0 +1,163 @@
+
+        // Generate timestamp for the export filename
+        const timestamp = new Date().toISOString().replace(/[:.]/g, '-')
+        const dumpFilename = `${dbCredentials.dbname || 'postgres'}_${timestamp}.sql`
+        const dumpFilePath = path.join(TMP_DIR, dumpFilename)
+        const plainSqlPath = path.join(TMP_DIR, `plain_${dumpFilename}`)
+        const sanitizedDumpPath = path.join(
+            TMP_DIR,
+            `sanitized_${dumpFilename}`
+        )
+
+        // Use pg_dump from the Lambda layer to dump the database
+        console.info(`Dumping database to ${dumpFilePath}...`)
+
+        try {
+            // Set environment variables for pg_dump
+            process.env.PGPASSWORD = dbCredentials.password
+
+            // Construct the pg_dump command
+            const pgDumpCmd = [
+                'pg_dump',
+                `-h ${dbCredentials.host}`,
+                `-p ${dbCredentials.port || 5432}`,
+                `-U ${dbCredentials.username}`,
+                `-d ${dbCredentials.dbname || 'postgres'}`,
+                '--format=plain',
+                `--file=${dumpFilePath}`,
+            ].join(' ')
+
+            // Execute pg_dump
+            execSync(pgDumpCmd, { stdio: 'inherit' })
+
+            console.info('Database dump completed successfully')
+
+            // Read, sanitize, and write back SQL content
+            console.info('Sanitizing email addresses...')
+            const sqlContent = fs.readFileSync(dumpFilePath, 'utf8')
+            const sanitizedContent = sanitizeSqlDump(sqlContent)
+            fs.writeFileSync(sanitizedDumpPath, sanitizedContent)
+        } catch (error) {
+            console.error('Error executing pg_dump:', error)
+            throw new Error(`pg_dump execution failed: ${error}`)
+        } finally {
+            // Clear password from environment
+            delete process.env.PGPASSWORD
+        }
+
+        // Upload the sanitized dump to S3
+        const s3Key = `${S3_PREFIX}/${dumpFilename}`
+        console.info(`Uploading dump to S3: s3://${S3_BUCKET}/${s3Key}`)
+        const fileContent = fs.readFileSync(sanitizedDumpPath)
+        await s3Client.send(
+            new PutObjectCommand({
+                Bucket: S3_BUCKET,
+                Key: s3Key,
+                Body: fileContent,
+                ContentType: 'application/octet-stream',
+            })
+        )
+
+        console.info('Upload to S3 completed successfully')
+
+        // Clean up temporary files
+        fs.unlinkSync(dumpFilePath)
+        fs.unlinkSync(sanitizedDumpPath)
+        console.info('Temporary files cleaned up')
+    } catch (err) {
+        console.error('Error in db export process:', err)
+        return {
+            statusCode: 500,
+            body: JSON.stringify({
+                message: 'Error during db export process',
+                error: String(err),
+            }),
+        }
+    }
+}
+
+function sanitizeEmail(email: string): string {
```

Comment:
Should we also replace the `username` too? I think there could be enough data here to identify the person by the username if it was a full name. For instance a state contact, you can trace it back to a person from that state government.

Although I'm unsure if any of these names and emails are PII as they could all be publicly available, but it's always nice to be cautious.